gnomad/assessment/validate_vep.py

```diff
@@ -0,0 +1,179 @@
+    """
+    logger.info(
+        "importing vep_HT on dataset: gnomAD_%s_%s_vep%s", release, dataset, vep_version
+    )
+    # check the input parameters
+    if release == "v4.0" and vep_version == "101" and dataset == "exomes":
+        logger.warning(
+            "gnomAD v4.0 EXOMES does not have v101 VEP annotation, please use 105"
+            " instead"
+        )
+
+    if release == "v4.0" and vep_version == "101" and dataset == "genomes":
+        logger.warning(
+            "gnomAD v4.0 GENOMES does not have v101 VEP annotation, please use 105"
+            " instead"
+        )
+
+    if release == "v3.1" and vep_version == "105" and dataset == "genomes":
+        logger.warning(
+            "gnomAD v3.1 GENOMES does not have v105 VEP annotation, please use 101"
+            " instead"
+        )
+
+    if release not in ("v4.0", "v3.1"):
+        logger.warning("release should be v4.0 or v3.1 only")
+
+    if vep_version not in ("101", "105"):
+        logger.warning("vep_version should be 101 or 105 only")
+
+    # define the path of the VEP-annotated HT
+    # TODO: changed this part and relevant function to import the vep HT
+    # resources, instead of hard-coding the path
+    if release == "v4.0" and dataset == "exomes":
+        vep_ht = hl.read_table(
+            "gs://gnomad/v4.0/annotations/exomes/gnomad.exomes.v4.0.vep.ht"
+        )
+    elif release == "v4.0" and dataset == "genomes":
+        vep_ht = hl.read_table(
+            "gs://gnomad/v4.0/annotations/genomes/gnomad.genomes.v4.0.vep.ht"
+        )
+    elif release == "v3.1" and dataset == "genomes":
+        vep_ht = hl.read_table(
+            "gs://gnomad/release/3.1.4/ht/genomes/gnomad.genomes.v3.1.4.sites.ht"
+        )
+    else:
+        logger.warning(
+            "currently only support v4.0 exomes, v4.0 genomes and v3.1 genomes"
+        )
+
+    vep_ht = vep_ht.select(vep_ht.vep.transcript_consequences)
+
+    # join two tables by interval and locus as index key
+    vep_ht = vep_ht.annotate(
+        interval_annotations=ht.index(vep_ht.locus, all_matches=True)
+    )
+
+    vep_ht = vep_ht.filter(hl.is_defined(vep_ht.interval_annotations))
+
+    vep_ht = vep_ht.annotate(
+        gene_stable_ID=vep_ht.interval_annotations.gene_stable_ID,
+        biotype=vep_ht.transcript_consequences.biotype,
+    )
+
+    # select only the gene_stable_ID and biotype to save space
+    vep_ht = vep_ht.select("gene_stable_ID", "biotype")
+
+    # explode the vep_ht by gene_stable_ID
+    vep_ht = vep_ht.explode(vep_ht.gene_stable_ID)
+
+    # count the number of total variants and "protein-coding" variants in each interval
+    count_ht = vep_ht.group_by(vep_ht.gene_stable_ID).aggregate(
+        all_variants=hl.agg.count(),
+        variants_in_pcg=hl.agg.count_where(vep_ht.biotype.contains("protein_coding")),
+    )
+
+    count_ht = count_ht.checkpoint("gs://gnomad-tmp-4day/count_tmp.ht", overwrite=True)
```

Comment:
Use hail's `new_temp_file`, import is: `from hail.utils.misc import new_temp_file`